contexts/authentication.tsx

```diff
@@ -125,12 +133,16 @@ export const AuthenticationProvider = ({ children }) => {
   useEffect(() => {
     if (!DAOService) return;
     
-    window?.ethereum?.on("accountsChanged", () => {
-      DAOService.connect()
-        .then(connected => {
-          if (connected) updateWalletAddress();
-        })
-        .catch(error => console.log("Failed to change account", error));
+    window?.ethereum?.on("accountsChanged", (accounts) => {
+      if (window.connectedAddress)
```

Comment:
`if !DAOService.web3Connection.connected` is what you're looking for.

Don't forget that accounts CAN BE EMPTY (when user disconnects from metamask)